kevm-pyk/src/tests/integration/test_conformance.py

```diff
@@ -57,66 +57,99 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
     assert pretty == GOLDEN
 
 
-def _skipped_tests() -> set[Path]:
-    def read_test_list(path: Path) -> list[Path]:
-        return [REPO_ROOT / test_path for test_path in path.read_text().splitlines()]
+def read_gst_file(gst_file: Path) -> list[str]:
+    with gst_file.open() as f:
+        gst_data = json.load(f)
+    return gst_data.keys()
+
 
-    slow_tests = read_test_list(REPO_ROOT / 'tests/slow.llvm')
-    failing_tests = read_test_list(REPO_ROOT / 'tests/failing.llvm')
+def _skipped_tests() -> set[tuple[Path, str]]:
+    slow_tests = read_csv_file(REPO_ROOT / 'tests/slow.llvm')
+    failing_tests = read_csv_file(REPO_ROOT / 'tests/failing.llvm')
     return set(slow_tests + failing_tests)
 
 
+VM_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests/VMTests'
+ALL_VM_TESTS_CSV: Final = REPO_ROOT / 'tests/all_vm_tests.csv'
+
+BCHAIN_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests'
+ALL_BCHAIN_TESTS_CSV: Final = REPO_ROOT / 'tests/all_bchain_tests.csv'
+
+
+def create_csv_files() -> None:
+    def _write_csv(target: Path, content: set[tuple[Path, str]]) -> None:
+        with open(target, 'w', newline='') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerows(sorted(content))
+
+    all_bchain_tests = {
+        (file.relative_to(TEST_DIR), test_id)
+        for file in BCHAIN_TEST_DIR.glob('**/*.json')
+        for test_id in read_gst_file(file)
+    }
+    _write_csv(ALL_BCHAIN_TESTS_CSV, all_bchain_tests)
+    all_vm_tests = {
+        (file.relative_to(TEST_DIR), test_id)
+        for file in VM_TEST_DIR.glob('*/*.json')
+        for test_id in read_gst_file(file)
+    }
+    _write_csv(ALL_VM_TESTS_CSV, all_vm_tests)
+
+
+def read_csv_file(csv_file: Path) -> list[tuple[Path, str]]:
+    with csv_file.open(newline='') as file:
+        reader = csv.reader(file)
+        return sorted([(Path(row[0]), row[1]) for row in reader])
```

Comment:
You can sort the files statically, then
```suggestion
def read_csv_file(csv_file: Path) -> tuple[tuple[Path, str], ...]:
    with csv_file.open(newline='') as file:
        reader = csv.reader(file)
        return tuple(Path(row[0]), row[1]) for row in reader)
```